File: android/src/main/java/com/vgretailersdk/scannedBalancePointsRequest.kt
package com.vgretailersdk

import com.google.gson.annotations.SerializedName

class ScannedBalancePointsRequest{
    @SerializedName("categories")
    var categories: Array<Int> = emptyArray()
    @SerializedName("subCategories")
    var subCategories: Array<Int> = emptyArray()
    @SerializedName("userId")
    var userId : String = ""
    @SerializedName("scheme")
    var scheme: Array<String> = emptyArray()
    constructor(categories: Array<Int>,subCategories: Array<Int>,userId: String, scheme: Array<String>) {
        this.userId = userId
        this.categories = categories
        this.subCategories = subCategories
        this.scheme = scheme
    }

}

File: ios/ScannedBalancePointsRequest.swift
import Foundation

class ScannedBalancePointsRequest: Codable {
    var categories: [Int]
    var subCategories: [Int]
    var userId: String
    var scheme:String
    // Custom initializer to initialize the object with values from the NSDictionary
    init(details: NSDictionary) {
        categories = details["categories"] as? [Int] ?? []
        subCategories = details["subCategories"] as? [Int] ?? []
        userId = details["userId"] as? String ?? ""
        scheme = details["scheme"] as? String ?? ""
    }
}

